### PR TITLE
Clean up optional dependency groups

### PR DESCRIPTION
**Pull request by dstansby**
Fixes https://github.com/zarr-developers/zarr-python/issues/2539.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -53,9 +53,25 @@ license = {text = "MIT License"}
...
+]
+gpu = [
+    "cupy-cuda12x",
+]
+universal-pathlib = [
+    'universal-pathlib>=0.0.22'
+]
+tree = [
+    'rich',
+]
```
Comment (jhamman) [inline pyproject.toml:73]:
```suggestion
rich = [
    'rich',
]
```

File: pyproject.toml [hunk 2]
```diff
@@ -68,15 +84,7 @@ test = [
...
-
-jupyter = [
-    'notebook',
-    'ipytree>=0.2.2',
-    'ipywidgets>=8.0.0',
-]
-gpu = [
-    "cupy-cuda12x",
-]
+optional = ["zarr[universal-pathlib, tree]"]
```
Comment (jhamman) [inline pyproject.toml:87]:
```suggestion
optional = ["zarr[universal-pathlib, rich]"]
```

**Review by jhamman — COMMENTED** [review]
Thanks. Just one suggestion. 

For groups with only one optional dependency, I think its best if we just use the dependency as the name.

**Comment by dstansby** [comment]
Is there any benefit in having a dependency group where the name is the same as the package? I'd suggest we either
- leave it as `tree`, to indicate "install this extra if you want to use `tree()`
-  remove it altogether and just document that folks need to install `rich` to use `tree()`

File: pyproject.toml [hunk 3]
```diff
@@ -53,9 +53,25 @@ license = {text = "MIT License"}
...
 [project.optional-dependencies]
+# User extras
 fsspec = [
     "fsspec>=2023.10.0",
 ]
+jupyter = [
+    'notebook',
+    'ipytree>=0.2.2',
+    'ipywidgets>=8.0.0',
+]
```
Comment (jhamman) [inline pyproject.toml:64]: I believe these can be removed.

**Review by jhamman — COMMENTED** [review]
> Is there any benefit in having a dependency group where the name is the same as the package? I'd suggest we either
> 
> * leave it as `tree`, to indicate "install this extra if you want to use `tree()`

If we do this, we should adopt the "feature-oriented groups" approach for the other groups as well. That would apply to fsspec for `RemoteStore`.

> * remove it altogether and just document that folks need to install `rich` to use `tree()`

I favor this option actually for `rich` and `universal-pathlib`.

File: pyproject.toml (hunk 3 above)
Comment (dstansby) [reply pyproject.toml]: 👍 done

**Comment by dstansby** [comment]
pre-commit.ci autofix

**Comment by dstansby** [comment]
> If we do this, we should adopt the "feature-oriented groups" approach for the other groups as well

👍 done. I also removed universal-pathlib, tree, and started a v3 what's new (I don't think this already exists?) to communicate what's changed from v2.

**Review by jhamman — APPROVED** [review]

**Pull request merged**